Internal Error occurred when formatting a file with multi-line KDoc comment

<!-- The bug you're experiencing might have already been reported!
Please search in the [issues](https://github.com/pinterest/ktlint/issues) before creating one. -->

## Expected Behavior
<!---Tell us what should happen. -->
<!-- If relevant, provide a link to the relevant section in the [Kotlin Coding Conventions](https://kotlinlang.org/docs/coding-conventions.html) or [Android Kotlin Style Guide](https://developer.android.com/kotlin/style-guide).  -->

<!-- Note: Ktlint should not produce any code which conflict with IntelliJ default code formatting. So please verify that your expectation is accepted by IntelliJ default code formatting. -->

## Observed Behavior
<!---Tell us what happens instead of the expected behavior -->
<!--- Provide the exact command which was executed but please -->
<!--- ensure that the flag "--log-level=debug" is added to the -->
<!--- command as well. Provide the output of this command. -->

An Internal Error occurred when formatting a file with multi-line KDoc comment.

The file I tried to format: 
```
class ClassA
/**
 * some comment 
 */(paramA: String)
```

Executed command & Error logs(stacktrace):
```
$ ktlint --format ClassA.kt
/tmp/ClassA.kt:0:0: Internal Error (rule 'standard:parameter-list-wrapping') in ClassA.kt at position '0:0. Please create a ticket at https://github.com/pinterest/ktlint/issues and provide the source code that triggered an error.
com.pinterest.ktlint.rule.engine.api.KtLintRuleException: Rule 'standard:parameter-list-wrapping' throws exception in file 'ClassA.kt' at position (0:0)
   Rule maintainer: KtLint
   Issue tracker  : https://github.com/pinterest/ktlint/issues
   Repository     : https://github.com/pinterest/ktlint
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRule(RuleExecutionContext.kt:65)
	at com.pinterest.ktlint.rule.engine.api.KtLintRuleEngine$format$3.invoke(KtLintRuleEngine.kt:146)
	at com.pinterest.ktlint.rule.engine.api.KtLintRuleEngine$format$3.invoke(KtLintRuleEngine.kt:145)
	at com.pinterest.ktlint.rule.engine.internal.VisitorProvider$visitor$3.invoke(VisitorProvider.kt:46)
	at com.pinterest.ktlint.rule.engine.internal.VisitorProvider$visitor$3.invoke(VisitorProvider.kt:44)
	at com.pinterest.ktlint.rule.engine.api.KtLintRuleEngine.format(KtLintRuleEngine.kt:145)
	at com.pinterest.ktlint.cli.internal.KtlintCommandLine.format(KtlintCommandLine.kt:485)
	at com.pinterest.ktlint.cli.internal.KtlintCommandLine.process(KtlintCommandLine.kt:471)
	at com.pinterest.ktlint.cli.internal.KtlintCommandLine.access$process(KtlintCommandLine.kt:57)
	at com.pinterest.ktlint.cli.internal.KtlintCommandLine$lintFiles$3.invoke$lambda$0(KtlintCommandLine.kt:411)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NullPointerException: Cannot invoke "org.jetbrains.kotlin.com.intellij.psi.impl.source.codeStyle.IndentHelper.getIndent(org.jetbrains.kotlin.com.intellij.psi.PsiFile, org.jetbrains.kotlin.com.intellij.lang.ASTNode)" because the return value of "org.jetbrains.kotlin.com.intellij.psi.impl.source.codeStyle.IndentHelper.getInstance()" is null
	at org.jetbrains.kotlin.com.intellij.psi.impl.source.codeStyle.CodeEditUtil.saveWhitespacesInfo(CodeEditUtil.java:108)
	at org.jetbrains.kotlin.com.intellij.psi.impl.source.codeStyle.CodeEditUtil.removeChildren(CodeEditUtil.java:120)
	at org.jetbrains.kotlin.com.intellij.psi.impl.source.codeStyle.CodeEditUtil.removeChild(CodeEditUtil.java:35)
	at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.CompositeElement.deleteChildInternal(CompositeElement.java:451)
	at org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement.delete(LeafPsiElement.java:182)
	at com.pinterest.ktlint.ruleset.standard.rules.ParameterListWrappingRule.wrapParameterInList(ParameterListWrappingRule.kt:241)
	at com.pinterest.ktlint.ruleset.standard.rules.ParameterListWrappingRule.wrapParameterList(ParameterListWrappingRule.kt:192)
	at com.pinterest.ktlint.ruleset.standard.rules.ParameterListWrappingRule.beforeVisitChildNodes(ParameterListWrappingRule.kt:79)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$1.invoke(RuleExecutionContext.kt:125)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$1.invoke(RuleExecutionContext.kt:124)
	at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle(SuppressHandler.kt:28)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:124)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:93)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.access$executeRuleOnNodeRecursively(RuleExecutionContext.kt:30)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:132)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:131)
	at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle(SuppressHandler.kt:28)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:131)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:93)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.access$executeRuleOnNodeRecursively(RuleExecutionContext.kt:30)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:132)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:131)
	at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle(SuppressHandler.kt:28)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:131)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:93)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.access$executeRuleOnNodeRecursively(RuleExecutionContext.kt:30)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:132)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext$executeRuleOnNodeRecursively$2$1.invoke(RuleExecutionContext.kt:131)
	at com.pinterest.ktlint.rule.engine.internal.SuppressHandler.handle(SuppressHandler.kt:28)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:131)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRuleOnNodeRecursively(RuleExecutionContext.kt:93)
	at com.pinterest.ktlint.rule.engine.internal.RuleExecutionContext.executeRule(RuleExecutionContext.kt:62)
	... 13 more
 ()

Summary error count (descending) by rule:
  An internal error occurred in the Ktlint Rule Engine: 1

```
## Steps to Reproduce
<!--- Provide a code example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Minimize the example to the bare -->
<!--- minimum required to reproduce the bug. Small, well written -->
<!--- example are proven to be resolved quicker. -->

1. Create this file.
```kotlin
class ClassA
/**
 * some comment
 */(paramA: String)
```
2. Execute `format` command.
```
$ ktlint --format ClassA.kt
```

### Additional information

If there is a space between the slash '/' at the end of the multi-line comment and the parentheses '(' at the beginning of the property definition, no internal error will occur and the format will succeed. The execution result will be as follows.

The file to format( a space between '/' and '(' ): 
```kotlin
class ClassA
/**
 * some comment
 */ (paramA: String)
```

Result:
```kotlin
class ClassA
/**
 * some comment
 */
(paramA: String)
```

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version of ktlint used: 1.1.1
* Relevant parts of the `.editorconfig` settings:
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): command line
* Version of Gradle used (if applicable):
* Operating System and version:

--- a/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KdocWrappingRule.kt
+++ b/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KdocWrappingRule.kt
@@ -14,7 +14,7 @@ import com.pinterest.ktlint.rule.engine.core.api.indent
 import com.pinterest.ktlint.rule.engine.core.api.nextLeaf
 import com.pinterest.ktlint.rule.engine.core.api.noNewLineInClosedRange
 import com.pinterest.ktlint.rule.engine.core.api.prevLeaf
-import com.pinterest.ktlint.rule.engine.core.api.upsertWhitespaceBeforeMe
+import com.pinterest.ktlint.rule.engine.core.api.upsertWhitespaceAfterMe
 import com.pinterest.ktlint.ruleset.standard.StandardRule
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
 
@@ -97,7 +97,7 @@ public class KdocWrappingRule :
     ) {
         emit(startOffset, "A KDoc comment may not be followed by any other element on that same line", true)
         if (autoCorrect) {
-            this.upsertWhitespaceBeforeMe(kdocCommentNode.indent())
+            kdocCommentNode.upsertWhitespaceAfterMe(kdocCommentNode.indent())
         }
     }
 

--- a/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ParameterListWrappingRule.kt
+++ b/ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ParameterListWrappingRule.kt
@@ -24,6 +24,8 @@ import com.pinterest.ktlint.rule.engine.core.api.editorconfig.INDENT_STYLE_PROPE
 import com.pinterest.ktlint.rule.engine.core.api.editorconfig.MAX_LINE_LENGTH_PROPERTY
 import com.pinterest.ktlint.rule.engine.core.api.firstChildLeafOrSelf
 import com.pinterest.ktlint.rule.engine.core.api.indent
+import com.pinterest.ktlint.rule.engine.core.api.isPartOfComment
+import com.pinterest.ktlint.rule.engine.core.api.isWhiteSpace
 import com.pinterest.ktlint.rule.engine.core.api.isWhiteSpaceWithNewline
 import com.pinterest.ktlint.rule.engine.core.api.leavesIncludingSelf
 import com.pinterest.ktlint.rule.engine.core.api.nextLeaf
@@ -74,16 +76,12 @@ public class ParameterListWrappingRule :
         emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
     ) {
         when (node.elementType) {
-            NULLABLE_TYPE -> wrapNullableType(node, emit, autoCorrect)
-            VALUE_PARAMETER_LIST -> {
-                if (node.needToWrapParameterList()) {
-                    wrapParameterList(node, emit, autoCorrect)
-                }
-            }
+            NULLABLE_TYPE -> visitNullableType(node, emit, autoCorrect)
+            VALUE_PARAMETER_LIST -> visitParameterList(node, emit, autoCorrect)
         }
     }
 
-    private fun wrapNullableType(
+    private fun visitNullableType(
         node: ASTNode,
         emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
         autoCorrect: Boolean,
@@ -183,15 +181,26 @@ public class ParameterListWrappingRule :
             .orEmpty()
             .any { it.elementType == ElementType.ANNOTATION_ENTRY }
 
-    private fun wrapParameterList(
+    private fun visitParameterList(
         node: ASTNode,
         emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
         autoCorrect: Boolean,
     ) {
-        node
-            .children()
-            .forEach { child -> wrapParameterInList(child, emit, autoCorrect) }
+        if (isPrecededByComment(node)) {
+            emit(node.startOffset, "Parameter list should not be preceded by a comment", false)
+        } else if (node.needToWrapParameterList()) {
+            node
+                .children()
+                .forEach { child -> wrapParameterInList(child, emit, autoCorrect) }
+        }
     }
+
+    private fun isPrecededByComment(node: ASTNode) =
+        node
+            .prevLeaf { !it.isWhiteSpace() }
+            ?.prevLeaf()
+            ?.isPartOfComment()
+            ?: false
 
     private fun intendedIndent(child: ASTNode): String =
         when {

--- a/ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KdocWrappingRuleTest.kt
+++ b/ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/KdocWrappingRuleTest.kt
@@ -111,4 +111,31 @@ class KdocWrappingRuleTest {
             .hasLintViolation(2, 29, "A KDoc comment may not be followed by any other element on that same line")
             .isFormattedAs(formattedCode)
     }
+
+    @Test
+    fun `Issue 2535 - Given a class with a KDoc on the primary constructor instead of on the class name`() {
+        val code =
+            // Code sample below contains a code smell. The KDoc is at the wrong position.
+            """
+            class ClassA
+                /**
+                 * some comment
+                 */(paramA: String)
+            """.trimIndent()
+        val formattedCode =
+            // Code sample below contains a code smell. The KDoc is at the wrong position. The formatted code is not intended to show
+            // that this code is well formatted according to good coding practices. It merely is used to validate that the newline is
+            // inserted at the correct position in the AST so that no exception will be thrown in the ParameterListWrappingRule.
+            """
+            class ClassA
+                /**
+                 * some comment
+                 */
+                (paramA: String)
+            """.trimIndent()
+        kdocWrappingRuleAssertThat(code)
+            .addAdditionalRuleProvider { ParameterListWrappingRule() }
+            .hasLintViolation(4, 8, "A KDoc comment may not be followed by any other element on that same line")
+            .isFormattedAs(formattedCode)
+    }
 }

--- a/ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ParameterListWrappingRuleTest.kt
+++ b/ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ParameterListWrappingRuleTest.kt
@@ -702,4 +702,32 @@ class ParameterListWrappingRuleTest {
                 LintViolation(3, 61, "Exceeded max line length (60)", false),
             ).hasNoLintViolationsExceptInAdditionalRules()
     }
+
+    @Test
+    fun `Issue 2535 - Given a class with a KDoc on the primary constructor instead of on the class name`() {
+        val code =
+            // Code sample below contains a code smell. The KDoc is at the wrong position.
+            """
+            class ClassA
+                /**
+                 * some comment
+                 */(paramA: String)
+            """.trimIndent()
+        val formattedCode =
+            // Code sample below contains a code smell. The KDoc is at the wrong position. The formatted code is not intended to show
+            // that this code is well formatted according to good coding practices. It merely is used to validate that the newline is
+            // inserted at the correct position in the AST so that no exception will be thrown in the ParameterListWrappingRule.
+            """
+            class ClassA
+            /**
+             * some comment
+             */
+            (paramA: String)
+            """.trimIndent()
+        parameterListWrappingRuleAssertThat(code)
+            .addAdditionalRuleProvider { KdocWrappingRule() }
+            .hasLintViolations(
+                LintViolation(4, 8, "Parameter list should not be preceded by a comment", false),
+            ).isFormattedAs(formattedCode)
+    }
 }

